controllers/mover/syncthing/syncthing.go

```diff
@@ -0,0 +1,399 @@
+	if err != nil {
+		return err
+	}
+	if err = json.Unmarshal(data, responseBody); err == nil {
+		st.SystemConnections = responseBody
+	}
+	return err
+}
+
+// jsonRequest performs a request to the Syncthing API and returns the response body.
+//nolint:funlen,lll,unparam,unused
+func (st *Syncthing) jsonRequest(endpoint string, method string, requestBody interface{}) ([]byte, error) {
+	// marshal above json body into a string
+	jsonBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return nil, err
+	}
+	// tostring the json body
+	body := io.Reader(bytes.NewReader(jsonBody))
+
+	// build new client if none exists
+	req, err := http.NewRequest(method, st.APIConfig.APIURL+endpoint, body)
+	if err != nil {
+		return nil, err
+	}
+
+	// set headers
+	headers, err := st.APIConfig.Headers()
+	if err != nil {
+		return nil, err
+	}
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	// make an HTTPS POST request
+	if err != nil {
+		return nil, err
+	}
+	resp, err := st.APIConfig.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// if there was an error, provide the information
+	if err := checkResponse(resp); err != nil {
+		return nil, err
+	}
+
+	// read body into response
+	return ioutil.ReadAll(resp.Body)
+}
+
+// SetEmptyAPIURL Sets the API URL for the Syncthing API, if one is not already defined.
+func (api *APIConfig) SetEmptyAPIURL(url string) {
+	if api.APIURL == "" {
+		api.APIURL = url
+	}
+}
+
+// Headers Returns a map containing the necessary headers for Syncthing API requests.
+// When no API Key is provided, an error is returned.
+func (api *APIConfig) Headers() (map[string]string, error) {
+	if api.APIKey == "" {
+		return nil, errors.New("API Key is not set")
+	}
+
+	return map[string]string{
+		"X-API-Key":    api.APIKey,
+		"Content-Type": "application/json",
+	}, nil
+}
+
+// BuildTLSClient Returns a new TLS client for Syncthing API requests.
+func (api *APIConfig) BuildOrUseExistingTLSClient() *http.Client {
+	if api.Client != nil {
+		return api.Client
+	}
+	return api.BuildTLSClient()
+}
+
+// BuildTLSClient Returns a new TLS client for Syncthing API requests.
+func (api *APIConfig) BuildTLSClient() *http.Client {
+	tlsConfig := api.TLSConfig
+	if tlsConfig == nil {
+		tlsConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		}
+	}
+
+	// load the TLS config with certificates
+	tr := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   time.Second * 5,
+	}
+	return client
+}
+
+// GenerateRandomBytes Generates random bytes of the given length using the OS's RNG.
+func GenerateRandomBytes(length int) ([]byte, error) {
+	// generates random bytes of given length
+	b := make([]byte, length)
+	_, err := rand.Read(b)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+// GenerateRandomString Generates a random string of the given length using the OS's RNG.
+func GenerateRandomString(length int) (string, error) {
```

Comment:
It's a documentation issue, but the function itself is not correct either. I rewrote this so that it is now only generating a random string of ascii characters which are mapped between values `[33, 126]` inclusive